### PR TITLE
Add preferences to ge dimmer switch device

### PR DESCRIPTION
**Pull request by NikitaKuchta**

### CODE REVIEW

**Comment by NikitaKuchta** [comment]
@dsobierajsk

**Comment by dsobierajsk** [comment]
New profile is needed since the change should apply for these 3 devices only:
```
  - id: 0063/4457
    deviceLabel: GE Dimmer Switch
    manufacturerId: 0x0063
    productType: 0x4457
    deviceProfileName: switch-level
  - id: 0063/4944
    deviceLabel: GE Dimmer Switch
    manufacturerId: 0x0063
    productType: 0x4944
    deviceProfileName: switch-level
  - id: 0063/5044
    deviceLabel: GE Dimmer Switch
    manufacturerId: 0x0063
    productType: 0x5044
    deviceProfileName: switch-level

**Review by dsobierajsk — CHANGES_REQUESTED** [review]
As mentioned, new profile is needed. Also preference should be added to preferences.lua file.

**Review by KKlimczukS — APPROVED** [review]
lgtm

**Review by dsobierajsk — CHANGES_REQUESTED** [review]

File: drivers/SmartThings/zwave-switch/profiles/ge-dimmer-switch.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  - name: "ledIndicator"
+    title: "LED Indicator"
+    description: "Turn LED indicator on..."
+    required: false
+    preferenceType: enumeration
+    definition:
+      options:
+        0: "on"
+        1: "off"
+        2: "never"
```
[the remark this answers is not in the document]
Comment (dsobierajsk) [reply drivers/SmartThings/zwave-switch/profiles/ge-dimmer-switch.yml:23]: The default value should be 0 also ;)

[the remark this answers is not in the document]
Comment (NikitaKuchta) [reply drivers/SmartThings/zwave-switch/profiles/ge-dimmer-switch.yml]: > The options should be:
> 
> ```
> 0: "off"
> 1: "on"
> 2: "never"
> ```
> 
> As it is in the DTH:
> 
> ```
> void indicatorWhenOn() {
> 	sendEvent(name: "indicatorStatus", value: "when on", displayed: false)
> 	sendHubCommand(new physicalgraph.device.HubAction(zwave.configurationV1.configurationSet(configurationValue: [1], parameterNumber: 3, size: 1).format()))
> }
> 
> void indicatorWhenOff() {
> 	sendEvent(name: "indicatorStatus", value: "when off", displayed: false)
> 	sendHubCommand(new physicalgraph.device.HubAction(zwave.configurationV1.configurationSet(configurationValue: [0], parameterNumber: 3, size: 1).format()))
> }
> 
> void indicatorNever() {
> 	sendEvent(name: "indicatorStatus", value: "never", displayed: false)
> 	sendHubCommand(new physicalgraph.device.HubAction(zwave.configurationV1.configurationSet(configurationValue: [2], parameterNumber: 3, size: 1).format()))
> }
> ```

right, done

**Review by dsobierajsk — APPROVED** [review]
lgtm

File: drivers/SmartThings/zwave-switch/profiles/ge-dimmer-switch.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  - name: "ledIndicator"
+    title: "LED Indicator"
+    description: "Turn LED indicator on..."
+    required: false
+    preferenceType: enumeration
+    definition:
+      options:
+        0: "off"
+        1: "on"
+        2: "never"
```
Comment (greens) [inline drivers/SmartThings/zwave-switch/profiles/ge-dimmer-switch.yml:23]: the text for the preference should match the text from the DTH, "on", "off", "never" are the keys for the enum, but the display values are "When Off", "When On", and "Never"

Comment (NikitaKuchta) [reply drivers/SmartThings/zwave-switch/profiles/ge-dimmer-switch.yml:23]: > the text for the preference should match the text from the DTH, "on", "off", "never" are the keys for the enum, but the display values are "When Off", "When On", and "Never"

I think that here is the place where the keys for the enum set:
`void indicatorWhenOn() {
	sendEvent(name: "indicatorStatus", value: "when on", displayed: false)
	sendHubCommand(new physicalgraph.device.HubAction(zwave.configurationV1.configurationSet(configurationValue: [1], parameterNumber: 3, size: 1).format()))
}

void indicatorWhenOff() {
	sendEvent(name: "indicatorStatus", value: "when off", displayed: false)
	sendHubCommand(new physicalgraph.device.HubAction(zwave.configurationV1.configurationSet(configurationValue: [0], parameterNumber: 3, size: 1).format()))
}

void indicatorNever() {
	sendEvent(name: "indicatorStatus", value: "never", displayed: false)
	sendHubCommand(new physicalgraph.device.HubAction(zwave.configurationV1.configurationSet(configurationValue: [2], parameterNumber: 3, size: 1).format()))
}`

Comment (greens) [reply drivers/SmartThings/zwave-switch/profiles/ge-dimmer-switch.yml:23]: It's actually here: https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/3cc8efa81eee0ac4c2a9ed420e1a6c25f88dedf0/devicetypes/smartthings/dimmer-switch.src/dimmer-switch.groovy#L49

that line is for a custom attribute

Comment (greens) [reply drivers/SmartThings/zwave-switch/profiles/ge-dimmer-switch.yml:23]: you can also see the keys (as opposed to the values) for enum being used here: https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/3cc8efa81eee0ac4c2a9ed420e1a6c25f88dedf0/devicetypes/smartthings/dimmer-switch.src/dimmer-switch.groovy#L92

off/on/never wouldn't display with the actual device

**Comment by greens** [comment]
@NikitaKuchta you added the preference for the devices in the Dimmer Switch DTH. I'm pointing out that the same issue exists for some devices on the Z-Wave Switch DTH. It would just make more sense to do them now rather than waiting for a bug to be opened about it.

**Review by dsobierajsk — APPROVED** [review]

File: drivers/SmartThings/zwave-switch/fingerprints.yml [hunk 3]
```diff
@@ -399,7 +399,7 @@ zwaveManufacturer:
     deviceLabel: Evolve Switch
     manufacturerId: 0x0113
     productType: 0x5257
-    deviceProfileName: switch-binary
+    deviceProfileName: dimmer-switch
```
Comment (greens) [inline drivers/SmartThings/zwave-switch/fingerprints.yml:402]: this device does not support Switch Level

also there are three other fingerprints that should have a new profile in addition to this one: https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/master/devicetypes/smartthings/zwave-switch.src/zwave-switch.groovy#L24

File: drivers/SmartThings/zwave-switch/profiles/dimmer-switch.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+name: dimmer-switch
```
Comment (greens) [inline drivers/SmartThings/zwave-switch/profiles/dimmer-switch.yml:1]: can you rename this profile to `switch-level-indicator`?

**Review by greens — APPROVED** [review]

**Pull request merged**